src/rules.ts

```diff
@@ -1658,6 +1659,168 @@ export const rules: Rule[] = [
+        } else {
+          resultAliasesArray = [title, ...resultAliasesArray];
+        }
+
+        const resultStyle = isSingleString ? (hasLinterMarkerComment ? 'Single string' : options['YAML aliases new array style']) : (isMultiline ? 'Multi-line array' : 'Single-line array');
+
+        let newAliasesYaml;
+
+        switch (resultStyle) {
+          case 'Multi-line array': {
+            const tailArrayYaml = resultAliasesArray.length === 1 ? '' : `\n${toYamlString(resultAliasesArray.slice(1))}`.replace(/\n-/g, '\n  -');
+            newAliasesYaml = `\n  - ${toYamlString(resultAliasesArray[0])} ${linterMarkerComment}${tailArrayYaml}`;
+            break;
+          }
+          case 'Single-line array':
+            newAliasesYaml = ` ${toSingleLineArrayYamlString(resultAliasesArray)} ${linterMarkerComment}`;
+            break;
+          case 'Single string':
+            newAliasesYaml = ` ${toYamlString(resultAliasesArray[0])} ${linterMarkerComment}`;
+            break;
+          default:
+            throw new Error(`Unsupported resultStyle: ${resultStyle}`);
+        }
+
+        text = text.replace(aliasesMatch[0], `aliases:${newAliasesYaml}\n`);
+
+        return text;
+      },
+      [
+        new Example(
+            'Adds a header with the title from heading.',
+            dedent`
+      # Obsidian
+      `,
+            dedent`
+      ---
+      aliases:
+        - Obsidian # linter-yaml-title-alias
+      ---
+      # Obsidian
+      `,
+            {
+              'YAML aliases new section style': 'Multi-line array',
+            },
+        ),
+        new Example(
+            'Adds a header with the title.',
+            dedent`
+      `,
+            dedent`
+      ---
+      aliases:
+        - Filename # linter-yaml-title-alias
+      ---
+
+      `,
+            {
+              'metadata: file name': 'Filename',
+              'YAML aliases new section style': 'Multi-line array',
+            },
+        ),
+      ],
+      [
+        new DropdownOption(
+            'YAML aliases new section style',
+            'The style of the newly created aliases YAML property',
+            'Multi-line array',
+            [
+              new DropdownRecord(
+                  'Multi-line array',
+                  '```aliases:\\n  - Title```',
+              ),
+              new DropdownRecord(
+                  'Single-line array',
+                  '```aliases: [Title]```',
+              ),
+              new DropdownRecord(
+                  'Single string',
+                  '```aliases: Title```',
+              ),
+            ],
+        ),
+        new DropdownOption(
+            'YAML aliases new array style',
+            'The style of the aliases YAML property for newly created arrays',
+            'Multi-line array',
+            [
+              new DropdownRecord(
+                  'Multi-line array',
+                  '```aliases:\\n  - Title```',
+              ),
```

Comment:
Why are there two options here? It looks like the first dropdown is sufficient.

Reply:
No, first dropdown is used when `aliases` section is created from scratch.

The second dropdown is needed in the following scenario

1. First dropdown is set to be `Single string`
2. The note already has `Single string` `aliases` section
`
---
aliases: alias1
---
`
3. Now the linter has to write a new alias from title. So now we have two aliases, so we need to write an array, and we need to define is it going to be single-line or multi-line. Because of point 1, it's unclear, so we need another dropdown
`
---
aliases: [Title, alias1] # linter-yaml-title-alias
---
`

vs

`
---
aliases:
  - Title # linter-yaml-title-alias
  - alias1
---
`

Reply:
I don't think I follow why you cannot use the current style for the update. When you have a single string it is spaces which dictate what is an alias. For example,
``` yaml
aliases: alias1 alias2
```
should be the same as 
``` yaml
aliases: 
 - alias1 
 - alias2
```

Reply:
In order to keep an alias that has spaces in it as one, it should be inside of quotes. But maybe Obsidian does not work this way. I believe that it did, but I could be wrong.

Reply:
I am sorry, but this is not the way YAML works, `alias1 alias2` is a single string, it is the same as

```
aliases:
  - alias1 alias2
```

Reply:
I tested it out and I must have seen things unless it was the tags section that allowed for weird functionality. I am aware that YAML does not allow it, but not all of what Obsidian does is valid YAML. That said, this is correct, I have no problem with this, but would like it if we could make it clear in the description of the drop down that the dropdown is meant for scenarios that have more than 1 alias.